src/main/java/nl/knmi/geoweb/backend/product/sigmet/Sigmet.java

```diff
@@ -159,51 +162,83 @@ public SigmetLevelPart(SigmetLevelUnit unit, int val) {
 			this.unit=unit;
 			this.value=val;
 		}
+
+		public String toTAC() {
+			if (this.unit==SigmetLevelUnit.FL) {
+				return "FL"+value;
+			}
+			if (this.unit==SigmetLevelUnit.FT) {
+				return value+"FT";
+			}
+			if (this.unit==SigmetLevelUnit.M) {
+				return value+"M";
+			}
+			return "";
+		}
 	}
 
 	@JsonInclude(Include.NON_NULL)
 	@Getter
 	public static class SigmetLevel {
-		SigmetLevelPart lev1;
-		SigmetLevelPart lev2;
+		SigmetLevelPart[]levels;
+		SigmetLevelMode mode;
+
 		public SigmetLevel(){};
-		public SigmetLevel(SigmetLevelPart lev1) {
-			this.lev1=lev1;
+		public SigmetLevel(SigmetLevelPart lev1, SigmetLevelMode mode) {
+			this.levels=new SigmetLevelPart[1];
+			this.levels[0]=lev1;
+			this.mode=mode;
 		}
-		public SigmetLevel(SigmetLevelPart lev1, SigmetLevelPart lev2) {
-			this.lev1=lev1;
-			this.lev2=lev2;
+		public SigmetLevel(SigmetLevelPart lev1, SigmetLevelPart lev2, SigmetLevelMode mode) {
+			levels=new SigmetLevelPart[2];
+			this.levels[0]=lev1;
+			this.levels[1]=lev2;
+			this.mode=mode;
 		}
 		public String toTAC() {
-			if (this.lev1 != null) {
-				if (this.lev2 != null) {
-					if (this.lev1.unit == this.lev2.unit) {
-						if (this.lev1.unit == SigmetLevelUnit.FL) {
-							return "FL" + this.lev1.value + "/" + this.lev2.value;
-						}
-					}
-					if (this.lev1.unit == SigmetLevelUnit.SFC) {
-						return "SFC/" + (this.lev2.unit == SigmetLevelUnit.FL ? "FL" + this.lev2.value : this.lev2.value + this.lev2.unit.toString().toUpperCase());
+			switch (this.mode) {
+			case BETW:
+				if ((this.levels[0] != null)  && (this.levels[1] != null)) {
+					if (this.levels[0].getUnit().equals(this.levels[1].getUnit())&&this.levels[0].getUnit().equals(SigmetLevelUnit.FL)){
+						return this.levels[0].toTAC() + "/" + this.levels[1].value;
 					}
-					
-					if (this.lev1.unit == SigmetLevelUnit.ABV) {
-						return "ABV " + (this.lev2.unit == SigmetLevelUnit.FL ? "FL" + this.lev2.value : this.lev2.value + this.lev2.unit.toString().toUpperCase());
-					}
-					if (this.lev1.unit == SigmetLevelUnit.TOP_ABV) {
-						return "TOP ABV " + (this.lev2.unit == SigmetLevelUnit.FL ? "FL" + this.lev2.value : this.lev2.value + this.lev2.unit.toString().toUpperCase());
-					}
-					if (this.lev1.unit == SigmetLevelUnit.TOP) {
-						return "TOP " + (this.lev2.unit == SigmetLevelUnit.FL ? "FL" + this.lev2.value : this.lev2.value + this.lev2.unit.toString().toUpperCase());
-					}
-
-				} else {
-					if (this.lev1.unit == SigmetLevelUnit.M || this.lev1.unit == SigmetLevelUnit.FT) {
-						return this.lev1.value + this.lev1.unit.toString().toUpperCase();
-					}
-					if (this.lev1.unit == SigmetLevelUnit.FL) {
-						return "FL" + this.lev1.value;
+				}
+				break;
+			case BETW_SFC:
+				if (this.levels[0] != null) {
+					if (this.levels[1].unit == SigmetLevelUnit.M || this.levels[1].unit == SigmetLevelUnit.FT) {
+						return "SFC/"+this.levels[1].value + this.levels[1].unit.toString().toUpperCase();
```

Comment:
Can better do this.levels[1].toTAC() such that the unit has the correct position